server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java

```diff
@@ -114,7 +118,7 @@ public BaseCacheResource(InvocationHelper invocationHelper) {
                }
             }
          }
-         return putInCache(responseBuilder, cache, key, data, ttl, idle);
+         return putInCache(responseBuilder, cache, key, data, ttl, idle, request.getSubject());
```

Comment:
No, this is wrong: the cache object passed in should already have the Subject applied (cache.withSubject(subject)) (RestCacheManager does this)

Reply:
ok